functional/FunctionalTest.java

```diff
@@ -3284,6 +3284,254 @@ public static void composeObject_test6() throws Exception {
     }
   }
 
+  /** Test: enableObjectLegalHold(String bucketName, String objectName, String versionId) */
+  public static void enableObjectLegalHold_test() throws Exception {
+    if (!mintEnv) {
+      System.out.println(
+          "Test: enableObjectLegalHold(String bucketName, String objectName, String versionId)");
+    }
+    long startTime = System.currentTimeMillis();
+    String bucketName = getRandomName();
+    String objectName = getRandomName();
+    try {
+      client.makeBucket(bucketName, null, true);
+      try (final InputStream is = new ContentInputStream(1 * KB)) {
+        client.putObject(bucketName, objectName, is, new PutObjectOptions(1 * KB, -1));
+      }
+
+      // Enable object legal hold.
+      client.enableObjectLegalHold(bucketName, objectName, "");
+
+      if (!client.isObjectLegalHoldEnabled(bucketName, objectName, "")) {
+        System.out.println(
+            "FAILED. expected Object Legal Hold Enabled to be : " + "True" + ", got: False ");
+
+        mintFailedLog(
+            "enableObjectLegalHold",
+            "(String bucketName, String objectName, String versionId)",
+            startTime,
+            "FAILED" + " >>> ",
+            " expected Object Legal Hold Enabled to be : " + "True" + ", got: False");
+      }
+
+      client.disableObjectLegalHold(bucketName, objectName, "");
+      mintSuccessLog(
+          "enableObjectLegalHold(String bucketName, String objectName, String versionId)",
+          null,
+          startTime);
+    } catch (Exception e) {
+      ErrorResponse errorResponse = null;
+      if (e instanceof ErrorResponseException) {
+        ErrorResponseException exp = (ErrorResponseException) e;
+        errorResponse = exp.errorResponse();
+      }
+
+      // Ignore NotImplemented error
+      if (errorResponse != null && errorResponse.errorCode() == ErrorCode.NOT_IMPLEMENTED) {
+        mintIgnoredLog(
+            "enableObjectLegalHold(String bucketName, String objectName, String versionId)",
+            null,
+            startTime);
+      } else {
+        mintFailedLog(
+            "enableObjectLegalHold(String bucketName, String objectName, String versionId)",
+            null,
+            startTime,
+            null,
+            e.toString() + " >>> " + Arrays.toString(e.getStackTrace()));
+        throw e;
+      }
+    } finally {
+      client.removeObject(bucketName, objectName);
+      client.removeBucket(bucketName);
```

Comment:
This is problematic to have an failure here. Below logic is better to handle corners.
```java
    try {
      client.makeBucket(...);
      try {
        client.putObject(...);
        client.enableObjectLegalHold(...);
        if (!client.isObjectLegalHoldEnabled(...)) {
          throw Exception(...);
        }
        client.disableObjectLegalHold(...);
        mintSucessLog(...);
      } finally {
        client.removeObject(...);
        client.removeBucket(...);
      }
    } catch (Exception e) {
      ErrorResponse errorResponse = null;
      if (e instanceof ErrorResponseException) {
        ErrorResponseException exp = (ErrorResponseException) e;
        errorResponse = exp.errorResponse();
      }

      // Ignore NotImplemented error
      if (errorResponse != null && errorResponse.errorCode() == ErrorCode.NOT_IMPLEMENTED) {
        mintIgnoredLog(
            "enableObjectLegalHold(String bucketName, String objectName, String versionId)",
            null,
            startTime);
      } else {
        mintFailedLog(
            "enableObjectLegalHold(String bucketName, String objectName, String versionId)",
            null,
            startTime,
            null,
            e.toString() + " >>> " + Arrays.toString(e.getStackTrace()));
        throw e;
      }
    }
```